src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -917,14 +897,20 @@ namespace winrt::TerminalApp::implementation
             // editors, who will write a temp file, then rename it to be the
             // actual file you wrote. So listen for that too.
             wil::FolderChangeEvents::FileName | wil::FolderChangeEvents::LastWriteTime,
-            [this, settingsBasename = settingsPath.filename(), stateBasename = statePath.filename()](wil::FolderChangeEvent, PCWSTR fileModified) {
-                const auto modifiedBasename = std::filesystem::path{ fileModified }.filename();
+            [this, settingsBasename = settingsPath.filename()](wil::FolderChangeEvent, PCWSTR fileModified) {
+                // DO NOT create a static reference to
+                // ApplicationState::SharedInstance here. See
+                // https://github.com/microsoft/terminal/pull/11222/files/9ff2775122a496fb8b1bcc7a0b83a64ce5b26c5f#r719627541
+                // for why. ApplicationState::SharedInstance already caches it's
+                // own static ref.
+
+                const winrt::hstring modifiedBasename{ std::filesystem::path{ fileModified }.filename().c_str() };

```

Comment:
minor performance concern -- this allocates an hstring (another copy) for every file that is ever changed in this folder. Is there something we can do to not do that?

Reply:
Eventually we'll have to pass the path to `ApplicationState::IsStatePath(...)`, which would need to take the `hstring`, since that lives in TSM.dll.

We could break the encapsulation and have `AppLogic` know what the app state paths are. Alternatively, we could _not really worry about someone writing to a bunch of files in this directory in rapid succession like a madman_ 🤔

Reply:
We should use `.native()` more liberally on `std::filesystem::path`s. It returns a constant reference to the underlying wide string for free.